pages/api/retrieveandquery.ts

```diff
@@ -26,14 +28,15 @@ type Output = {
 
 export default async function handler(
   req: NextApiRequest,
-  res: NextApiResponse<Output>
+  res: NextApiResponse<Output>,
 ) {
   if (req.method !== "POST") {
     res.status(405).json({ error: "Method not allowed" });
     return;
   }
 
-  const { query, topK, nodesWithEmbedding }: Input = req.body;
+  const { query, topK, nodesWithEmbedding, temperature, topT }: Input =
```

Comment:
Thanks for doing the PR! A couple of things: it's top P, not top T, https://community.openai.com/t/a-better-explanation-of-top-p/2426

And you'll want to initialize a new OpenAI class and pass in the temperature and top P. An example is here: https://github.com/run-llama/LlamaIndexTS/blob/main/apps/simple/vectorIndexAnthropic.ts